### DISCUSSION TITLE
Can this be done in prod?

### DISCUSSION THREAD

**Discussion by vosechu** [mitmproxy/mitmproxy#5351] [Q&A]
Just saw the tool mitmproxy2swagger, and I wonder if there’s any way to do mitmproxy in prod to capture flows and then export them. 

Use case: you’ve just inherited an api from another team, and you need to understand how it _really_ works, not just how they _say_ it works. How the customers actually use it. So it’s critical to be able to capture flows from real customers, somehow merging the traffic, and building a map of actual use. 

We’ve done this with a custom in-house tool exporting to an object store, but this looks like it would do the job even better. 

What do you think, could this be run in production somehow? Thanks!

**Comment by mhils** [comment]
This depends a bit on what your production looks like. Performance-wise, mitmproxy will happily handle 100 req/s, but we're of course far off from nginx/Apache speeds. You likely want to run mitmdump, which is the same as mitmproxy, but headless (without the UI). Very often it makes sense to run it as a [reverse proxy](https://docs.mitmproxy.org/stable/concepts-modes/). Depending on your use case you may want to enable request/response streaming.

I know a whole bunch of companies that happily run mitmproxy in production, and there are very likely many more which I don't know about. The best way really is to try it out and report back how it went. :-)

**Reply by vosechu** [reply]
Awesome! Thank you so much!!

Our total production volume is much, much higher, but I think that we could deploy a special project and limit it in the load balancer. 

Thank you for teaching me about mitmdump! I’ll look that up and learn more!

I also appreciate knowing that there are other folks using this in prod! That helps a bunch!
